docs/csharp/ref-locals-and-returns.md

```diff
@@ -0,0 +1,205 @@
+so each argument is 24 bytes. By specifying the `in` modifier, you pass a
+reference to those arguments, either 4 bytes or 8 bytes each depending on the
+architecture of the machine. The difference in size is small, but it can quickly add
+up when your application calls this method in a tight loop using many different
+values.
+ 
+The `in` modifier complements `out` and `ref` in other ways as well. You
+cannot create overloads of a method that differ only in the presence of
+`in`, `out` or `ref`. These new rules extend the same behavior that had always been
+defined for `out` and `ref` parameters.
+
+The `in` modifier may be applied to any member that takes parameters:
+methods, delegates, lambdas, local functions, indexers, operators.
+
+Unlike `ref` and `out` arguments, you may use literal values or constants
+for the argument to an `in` parameter. Also, unlike a `ref` or `out` parameter,
+you don't need to apply the `in` modifier at the call site. The following
+code shows you two examples of calling the `CalculateDistance` method. The
+first uses two local variables, passed by reference. The second include a
+temporary variable created as part of the method call. 
+
+[!code-csharp[UseInArgument](../../samples/csharp/reference-semantics/Program.cs#UseInArgument "Specifying an In argument")]
+
+There are several ways in which the compiler ensures that the readonly
+nature of an `in` argument is enforced.  First of all, the called method
+can't directly assign to an `in` parameter. It can't directly assign
+to any field of an `in` parameter. In addition, you cannot pass
+an `in` parameter to any method demanding the `ref` or `out` modifier.
+The compiler enforces that the `in` argument is a readonly variable. You
+can call any instance method that uses pass-by-value semantics. In
+those instances, a copy of the `in` parameter is created. Because the compiler
+can create a temporary for any `in` parameter, you can also specify default
+values for any `in` parameter. The follow code uses that to specify the origin
+as the default value for the second point:
+
+[!code-csharp[InArgumentDefault](../../samples/csharp/reference-semantics/Program.cs#InArgumentDefault "Specifying defaults for an in parameter")]
+
+The `in` parameter designation can also be used with reference types, or built in
+numeric values. However, the benefits inn both cases are minimal if any.
+
+## `ref readonly` returns
+
+You may also want to return a value type by reference, but disallow the
+caller from modifying that value. Use the `ref readonly` modifier to
+express that design intent. It notifies readers
+that you are returning a reference to existing data, but not allowing
+modification. 
+
+The compiler enforces that the caller cannot modify the reference. Attempts
+to assign to the value directly generate a compile-time error. Calling member
+methods creates a defensive copy. 
+
+Creating a copy of a ref readonly return is easy: Just assign it to a variable
```

Comment:
I'd move this to follow the example. It's best to have the example -- which is a good one -- follow immediately to provide a use case for ref readonly and make it less abstract.